### PR TITLE
Fix ProductVariant with duplicated AttributeValues

### LINKED ISSUE #4764: ProductVariants with duplicated AttributeValue
Mutation `ProductVariantCreate` allows creating `ProductVariant` with duplicated `AttributeValue`. It's shouldn't be avable to create product variant with the same values in attributes.

### PR DESCRIPTION
**Pull request by fowczarek**
I want to merge this change because resolve #4764 

### Pull Request Checklist

1. [ ] Privileged views and APIs are guarded by proper permission checks.
1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Database queries are optimized and the number of queries is constant.
1. [x] Database migration files are up to date.
1. [x] The changes are tested.
1. [x] GraphQL schema and type definitions are up to date.
1. [x] Changes are mentioned in the changelog.

### CODE REVIEW

File: tests/api/test_product.py [hunk 1]
```diff
@@ -1378,13 +1355,15 @@ def test_update_product_with_non_existing_attribute(
     # Try to assign multiple values from an attribute that does not support such things
     variables = {
         "productId": graphene.Node.to_global_id("Product", product.pk),
-        "attributes": attributes_input,
+        "attributes": [{"id": "QXR0cmlidXRlOjA=", "values": ["hello"]}],
```
Comment (NyanKiyoshi) [inline tests/api/test_product.py]: Hardcoded?

**Comment by django-queries** [comment]
Here is the report for 372ab733ea07033ae7e3567837ebb88dd4f4ad28 (mirumee/saleor @ 4764/duplicated_product_variants)
Base comparison is d6b06fc92beab0814328bc4cfe40a2c7470bda57.



<details><summary>No differences were found. (click me)</summary>
<p>

```diff
# api.benchmark checkout
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  add billing address to checkout            	         34	         34	             20
  add shipping to checkout                   	          7	          7	              0
  checkout payment charge                    	         14	         14	              0
  complete checkout                          	          6	          6	              0
  create checkout                            	         48	         48	             24

# api.benchmark homepage
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  retrieve main menu                         	          5	          5	              0
  retrieve product list                      	          4	          4	              0
  retrieve secondary menu                    	          5	          5	              0
  retrieve shop                              	          2	          2	              0

# api.benchmark product
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  product details                            	         15	         15	              3
  retrieve product attributes                	         13	         13	              2

# api.benchmark variant
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  retrieve variant list                      	         15	         15	              6

# api product sorting attributes
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  sort product not having attribute data     	         21	         21	              0
```

</p>
</details>

File: tests/api/test_product.py [hunk 2]
```diff
@@ -1333,43 +1333,26 @@ def test_update_product_without_supplying_required_product_attribute(
...
 def test_update_product_with_non_existing_attribute(
-    staff_api_client,
-    product,
-    permission_manage_products,
-    color_attribute,
-    attributes_input,
-    expected_message,
+    staff_api_client, product, permission_manage_products, color_attribute
 ):
+    non_existent_attribute_pk = 999999
```
Comment (NyanKiyoshi) [inline tests/api/test_product.py]:
```suggestion
    non_existent_attribute_pk = -1
```

File: tests/api/test_variant.py [hunk 3]
```diff
@@ -887,69 +849,38 @@ def test_product_variant_bulk_create_duplicated_sku_in_input(
...
-    assert product_variant_count == ProductVariant.objects.count()
-
-
 def test_product_variant_bulk_create_many_errors(
     staff_api_client, product, size_attribute, permission_manage_products
 ):
     product_variant_count = ProductVariant.objects.count()
     product_id = graphene.Node.to_global_id("Product", product.pk)
+    size_attribute_id = graphene.Node.to_global_id("Attribute", size_attribute.pk)
+    non_existent_attribute_pk = 999999
```
Comment (NyanKiyoshi) [inline tests/api/test_variant.py]:
```suggestion
    non_existent_attribute_pk = -1
```

File: saleor/graphql/product/utils.py [hunk 4]
```diff
@@ -45,3 +47,17 @@ def validate_attribute_input_for_variant(instance: models.Attribute, values: Lis
         raise ValidationError(
             "Attribute values cannot be blank", code=ProductErrorCode.REQUIRED
         )
+
+
+def get_used_attribute_values(product):
```
Comment (NyanKiyoshi) [inline saleor/graphql/product/utils.py]:
```suggestion
def get_used_variants_attribute_values(product):
```

**Pull request merged**